helm/pv-migrate/templates/rsync/job.yaml

```
@@ -38,8 +38,12 @@ spec:
               rc=1
               retries={{ .Values.rsync.maxRetries }}
               period={{ .Values.rsync.retryPeriodSeconds }}
-              {{ if .Values.rsync.fixPrivateKeyPerms -}}
-              chmod 400 {{ .Values.rsync.privateKeyMountPath }}
+              {{ if .Values.rsync.privateKeyMount -}}
+              privateKeyFilename=$(basename {{ .Values.rsync.privateKeyMountPath }})
+              mkdir -p ~/.ssh
+              chmod 700 ~/.ssh
+              cp -v {{ .Values.rsync.privateKeyMountPath }} ~/.ssh/
+              chmod 400 ~/.ssh/$privateKeyFilename
```

Quotes are probably necessary to avoid potential issues with spaces

```suggestion
              {{ if .Values.rsync.privateKeyMount -}}
              privateKeyFilename=$(basename "{{ .Values.rsync.privateKeyMountPath }}")
              mkdir -p ~/.ssh
              chmod 700 ~/.ssh
              cp -v "{{ .Values.rsync.privateKeyMountPath }}" ~/.ssh/
              chmod 400 "~/.ssh/$privateKeyFilename"
```